src/Model/Model.ts

```diff
@@ -1,55 +1,77 @@
 import { v4 as uuidv4 } from 'uuid';
-import { EventEmitter } from 'events';
+import { type Context } from './contexts';
+import { RacerBackend } from '../Backend';
+import { type Connection } from './connection';
+import { type ModelData } from './collections';
+import { Primitive } from '../types';
 
-interface DebugOptions {
-  debugMutations?: boolean,
-  disableSubmit?: boolean,
-}
+export type UUID = string;
+
+export type DefualtType = unknown;
+
+declare module './Model' {
+  interface DebugOptions {
+    debugMutations?: boolean,
+    disableSubmit?: boolean,
+    remoteMutations?: boolean,
+  }
+  
+  interface ModelOptions {
+    debug?: DebugOptions;
+    fetchOnly?: boolean;
+    unloadDelay?: number;
+    bundleTimeout?: number;
+  }
 
-interface ModelOptions {
-  debug?: DebugOptions;
+  type ErrorCallback = (err?: Error) => void;
 }
 
-type ModelInitFunction = (instance: Model, options: ModelOptions) => void;
+type ModelInitFunction = (instance: RootModel, options: ModelOptions) => void;
 
-export class Model extends EventEmitter {
+export class Model<T = DefualtType> {
   static INITS: ModelInitFunction[] = [];
 
   ChildModel = ChildModel;
   debug: DebugOptions;
-  root: Model;
+  root: RootModel;
+  data: T;
 
-  _at: () => Model;
-  _context: {};
+  _at: string;
+  _context: Context;
   _eventContext: number | null;
   _events: [];
   _maxListeners: number;
-  _pass: () => void;
-  _preventCompose: () => void;
+  _pass: any;
+  _preventCompose: boolean;
   _silent: boolean;
 
+
+  id(): UUID {
+    return uuidv4();
+  }
+
+  _child() {
+    return new ChildModel(this);
+  };
+}
+
+export class RootModel extends Model<ModelData> {
+  backend: RacerBackend;
```

Comment:
Technically should be optional, as it's only present on server-side models created with `backend.createModel()`, but may cause type breakages if made optional. Could try it and see